File: cs/src/ConstraintSolver/CNSAT/Types/Watcher.cs
using System;

namespace Alica.Reasoner.CNSAT
{
	public class Watcher
	{
		public Clause Clause{ get; set; } 
		public Lit Lit{	get; set; } 
		
		public Watcher (Lit l, Clause parent)
		{
			this.Clause = parent;
			this.Lit = l;
			Lit.Var.WatchList.Add(this);
		}
		
		
	}
}

File: java/src/main/java/Reasoner/CNSAT/Watcher.java
//
// Translated by CS2J (http://www.cs2j.com): 12.11.2016 09:40:46
//

package Reasoner.CNSAT;

public class Watcher
{
    private Clause Clause;
    public Clause getClause() {
        return Clause;
    }

    public void setClause(Clause value) {
        Clause = value;
    }

    private Lit Lit;
    public Lit getLit() {
        return Lit;
    }

    public void setLit(Lit value) {
        Lit = value;
    }

    public Watcher(Lit l, Clause parent) throws Exception {
        this.setClause(parent);
        this.setLit(l);
        getLit().getVar().getWatchList().add(this);
    }

}


